File: 1052-grumpy-bookstore-owner/1052-grumpy-bookstore-owner.py
class Solution:
    def maxSatisfied(self, customers: List[int], grumpy: List[int], minutes: int) -> int:
        grumpy = list(zip(grumpy,customers))
        res = 0

        for g,c in grumpy:
            if g == 0:
                res+=c

        l,r=0,minutes
        for i in range(minutes):
            if grumpy[i][0] == 1:
                res+=grumpy[i][1]
        curr = res
        while r < len(grumpy):
            if grumpy[l][0] == 1:
                curr-=grumpy[l][1]
            if grumpy[r][0] == 1:
                curr+=grumpy[r][1]
            res = max(curr,res)
            l+=1
            r+=1

        return res

        

File: 1052-grumpy-bookstore-owner/1052-grumpy-bookstore-owner.dart
class Solution {
  int maxSatisfied(List<int> customers, List<int> grumpy, int minutes) {
    int currSum = 0;
    int n = customers.length;
    for(int i = 0;i < n;i++){
        if(grumpy[i] == 0){
            currSum+=customers[i];
        }
    }
    int res = 0;
    for(int r = 0;r<n;r++){
        if(grumpy[r]==1){
            currSum+=customers[r];
        }
        int l = r-minutes;
        if(l>=0 && grumpy[l]==1){
            currSum-=customers[l];
        }
        res = max(res,currSum);
    }
    return res; 
  }
}